client/signup/controller.js

```diff
@@ -331,24 +331,28 @@ export default {
 		const { getState, dispatch } = signupStore;
 		const signupDependencies = getSignupDependencyStore( getState() );
 
-		const siteSlug = signupDependencies?.siteSlug || query?.siteSlug;
-		if ( ! siteSlug ) {
+		const siteIdOrSlug =
+			signupDependencies?.siteSlug ||
+			query?.siteSlug ||
+			signupDependencies?.siteId ||
+			query?.siteId;
+		if ( ! siteIdOrSlug ) {
 			next();
 			return;
 		}
-		const siteId = getSiteId( getState(), siteSlug );
+		const siteId = getSiteId( getState(), siteIdOrSlug );
 		if ( siteId ) {
 			dispatch( setSelectedSiteId( siteId ) );
 			next();
 		} else {
-			// Fetch the site by siteSlug and then try to select again
-			dispatch( requestSite( siteSlug ) )
+			// Fetch the site by siteSlugOrId and then try to select again
```

Comment:
`siteIdOrSlug` rather than ~siteSlugOrId~ 🤔

Reply:
Whoops! I changed by mind on the variable name half way through but forgot to update everywhere.
[Selectors like this one](https://github.com/Automattic/wp-calypso/blob/abdf35662927f9919535b675af4b2b2c7bcf5df4/client/state/sites/selectors/get-site-id.js#L18) call it `siteIdOrSlug` so I switched to using that order.